cmd/prometheus/main.go

```diff
@@ -425,11 +439,15 @@ func main() {
 	serverOnlyFlag(a, "storage.tsdb.allow-overlapping-compaction", "Allow compaction of overlapping blocks. If set to false, TSDB stops vertical compaction and leaves overlapping blocks there. The use case is to let another component handle the compaction of overlapping blocks.").
 		Default("true").Hidden().BoolVar(&cfg.tsdb.EnableOverlappingCompaction)
 
-	serverOnlyFlag(a, "storage.tsdb.wal-compression", "Compress the tsdb WAL.").
-		Hidden().Default("true").BoolVar(&cfg.tsdb.WALCompression)
+	var (
+		tsdbWALCompression     bool
+		tsdbWALCompressionType string
+	)
+	serverOnlyFlag(a, "storage.tsdb.wal-compression", "Compress the tsdb WAL. If false, the --storage.tsdb.wal-compression-type flag is ignored.").
+		Hidden().Default("true").BoolVar(&tsdbWALCompression)
 
-	serverOnlyFlag(a, "storage.tsdb.wal-compression-type", "Compression algorithm for the tsdb WAL.").
-		Hidden().Default(string(wlog.CompressionSnappy)).EnumVar(&cfg.tsdb.WALCompressionType, string(wlog.CompressionSnappy), string(wlog.CompressionZstd))
+	serverOnlyFlag(a, "storage.tsdb.wal-compression-type", "Compression algorithm for the tsdb WAL, used when --storage.tsdb.wal-compression is true.").
+		Hidden().Default(compression.Snappy).EnumVar(&tsdbWALCompressionType, compression.Types()...)
```

Comment:
Does this mean you can now choose `None` as the compression algorithm now? Looks like an accidental collateral 🤔

Reply:
```suggestion
	serverOnlyFlag(a, "storage.tsdb.wal-compression-type", "Compression algorithm for the tsdb WAL, used when --storage.tsdb.wal-compression is true.").
		Hidden().Default(compression.Snappy).EnumVar(&tsdbWALCompressionType, compression.Snappy, compression.Zstd)
```

Reply:
Resolved by the comment below.